frontend/src/employee-frontend/components/finance-basics/VoucherValueEditor.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useCallback, useState } from 'react'
+
+// import { FormState } from './ServiceNeedItem'
+import DateRange from 'lib-common/date-range'
+import { throwIfNull } from 'lib-common/form-validation'
+import { ServiceNeedOptionVoucherValueRange } from 'lib-common/generated/api-types/invoicing'
+import { parseCentsOrThrow } from 'lib-common/money'
+import { useMutationResult } from 'lib-common/query'
+import AsyncButton from 'lib-components/atoms/buttons/AsyncButton'
+import Button from 'lib-components/atoms/buttons/Button'
+import InputField from 'lib-components/atoms/form/InputField'
+import { Td, Tr } from 'lib-components/layout/Table'
+import DatePicker from 'lib-components/molecules/date-picker/DatePicker'
+import { Translations } from 'lib-customizations/employee'
+
+import { FormState } from './ServiceNeedItem'
+import { createVoucherValueMutation } from './queries'
+
+export type VoucherValueEditorProps = {
+  i18n: Translations
+  initialState: FormState
+  close: () => void
+}
+
+export default React.memo(function VoucherValueEditor({
+  i18n,
+  initialState,
+  close
+}: VoucherValueEditorProps) {
+  const [editorState, setEditorState] = useState<FormState>(initialState)
```

Comment:
All of the state handling and form validation here is kinda legacy now 😅 But I don't know if I'd start changing it now anymore. Maybe just keep in mind that if there's need for nicer looking validation or other major changes then refactoring to use form framework first is adviced. It's actually a nice tool once you get the hang of it, but usually needs some help to get pass the initial learning curve.